docs/components/popovers/popover-demo.js

```diff
@@ -0,0 +1,47 @@
+if (document.getElementById('vue-popoverDemo')) {
+    var PopoverComponent = Vue.component('popover-demo', {
```

Comment:
Updated the demo to use a Vue component to get better behavior.

Previously, the `_initialPosition` logic of the popover component would reset the position of the demo popover to `bottom-right` after each position change. 

For example, a user could change the demo popover position to `top-left`, but after opening and closing the popover, the position resets to `bottom-right`, even if the position dropdown selector still says `top-left`.

Now, the `hx-popover` is reinitialized every time the dropdown is changed, so that the `_initialPosition` of the demo popover is set to whatever is selected in the position dropdown.